go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +49,134 @@ func TestExpansionCosts(t *testing.T) {
 		}
 	}
 }
+
+func TestMemory_expandMemoryWithoutCharging(t *testing.T) {
+
+	test := map[string]struct {
+		size        uint64
+		initialMem  []byte
+		expectedMem []byte
+	}{
+		"empty-memory-increases-to-desired-size": {
+			size:        32,
+			initialMem:  []byte{},
+			expectedMem: []byte{31: 0x0},
+		},
+		"memory-bigger-than-size-changes-nothing": {
+			size:        32,
+			initialMem:  []byte{63: 0x0},
+			expectedMem: []byte{63: 0x0},
+		},
+		"size-zero-changes-nothing": {
+			size:        0,
+			initialMem:  []byte{31: 0x0},
+			expectedMem: []byte{31: 0x0},
```

Comment:
This cases stresses the same thing as the case above since the memory is greater than the size. Maybe change the `initialMem` to `[]byte{}`.